source/extensions/filters/http/composite/filter.cc

```diff
@@ -0,0 +1,188 @@
+#include "extensions/filters/http/composite/filter.h"
+
+#include "envoy/http/filter.h"
+
+#include "common/common/stl_helpers.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace Composite {
+
+namespace {
+template <class FilterPtrT, class FuncT, class RValT, class... Args>
+RValT delegateFilterAction(FilterPtrT& filter, FuncT func, RValT rval, Args&&... args) {
```

Comment:
nit: comments please and/or longer template class names, this type of code is really hard to read.